python/cudf/cudf/_lib/join.pyx

```diff
@@ -72,19 +72,21 @@ cpdef join(col_lhs, col_rhs, left_on, right_on, how, method):
 
     cdef cudf_table result
 
-    with nogil:
-        if how == 'left':
-            result = left_join(
-                list_lhs[0],
-                list_rhs[0],
-                left_on_ind,
-                right_on_ind,
-                columns_in_common,
-                <cudf_table*> NULL,
-                context
-            )
 
-        elif how == 'inner':
+    if how == 'left':
```

Comment:
Why are these outside the nogil now?

Reply:
For some reason the Cython compiler complained about not being able to do the comparison `how == "left"` without the GIL and I was deep in something else and didn't look too much into it. I'll try with the GIL again and debug

Reply:
Done. Weird.